enhancements/insights/config-api.md

```diff
@@ -0,0 +1,245 @@
+  - cons
+    - need to be able to interrupt gathering (if any in progress)
+
+The second option of forcing it is preferred and it also corresponds to the API proposal in the [Proposal](#proposal) section.  
+
+### Open Questions
+
+- ~~Can the API define default values? How does it work or how is it implemented?~~ This is done by using `// +kubebuilder:default=`
+- ~~Should the new time of forced gathering (if any defined by user) set a new time for the existing gathering interval? If I force gathering at time A, will the new automatic gathering be in time A + interval?~~ - yes. It makes sense to update the last known time of gathering so that the regular gathering does not happen immediately after the forced gathering. 
+
+### Test Plan
+
+The new config API will require updates of the existing Insights operator testsuite. The testsuite will be extended to cover the new API and respective attributes. 
+Following behavior will need to be tested:
+- `support` secret config takes precedence over the new config API
+- forced gathering works as expected
+- disabling some gatherers works as expected and diasbling all will disable the gathering completely
+
+### Graduation Criteria
+
+The plan is to introduce the first version of the new API as the `v1alpha1` version and it will be behind `TechPreviewNoUpgrade` feature gate. We would like to
+keep the original way of configuring the Insights operator (that is via `support` secret), because it is problematic to move every existing configuration option to the new configuration API and the existing `support` secret serves basically as a configuration API. 
+
+#### Dev Preview -> Tech Preview
+
+The original way of configuring the Insights is still available. If there is a conflict, in other words, if there is some configuration option defined in the `support` secret as well as in the new configuration API then the value from the `support` secret takes precedence. 
+
+#### Tech Preview -> GA
+
+The `TechPreviewNoUpgrade` feature gate requirement is removed. The behavior defined above in the [Dev Preview -> Tech Preview](#dev-preview---tech-preview) section
+is still true. 
+
+#### Removing a deprecated feature
+
+The existing configuration via `support` secret should be still available for backward compatibility reasons. The reason of the `support` secret existence is the option of using basic authentication for the Ingress service, which Insights operator uses as the upload endpoint. 
+
+### Upgrade / Downgrade Strategy
+
+No update of the configuration is required. The existing configuration in the `support` secret continues to work. When there is an existing `insights.config.openshift.io` kind in a cluster after upgrade, then the user can optionally use the new configuration API. 
+There can be a conflict when a configuration option is defined in the `support` secret as well as in the configuration API. The configuration in the `support` secret takes precedence in this case. 
```

Comment:
Worth providing a way to complain in `status` for "I realize `spec` asks for `$THING`, but the `support` Secret is blocking that"?  Or can we just assume that conflicts will be rare enough that it's not worth polishing that particular UX that much?

Reply:
Currently there's only one conflicting attribute and that is the `EnableObfuscation` so I think it will be rare (TBH I think using the `support` secret is rare today as well, but I don't have the data to support this assumption).

Reply:
As mentioned the conflicts will be probably rare. The conflict will be mentioned in the logs (and it should be well documented) and let's see.